src/models/Unit.ts

```diff
@@ -0,0 +1,51 @@
+import { UnitAttributes } from "./UnitAttributes";
+import { Action } from "../actions/Action";
+
+export abstract class Unit {
+  attributes: UnitAttributes;
+  actions: Action[] = [];
+
+  constructor(attributes: UnitAttributes) {
+    this.attributes = attributes;
+    this.initActions();
+  }
+
+  abstract getPossibleTargets(
+    battlefield: (Unit | null)[][],
+    enemyUnits: Unit[],
+  ): Unit[];
+
+  abstract initActions(): void;
+
+  getEnemyUnits(battlefield: (Unit | null)[][]): Unit[] {
+    const enemyUnits: Unit[] = [];
+    battlefield.forEach((row) => {
+      row.forEach((cell) => {
+        if (
+          cell &&
+          cell.attributes.team !== this.attributes.team &&
+          cell.attributes.hp > 0
+        ) {
+          enemyUnits.push(cell);
+        }
```

Comment:
From this moment think not only how to write a good code but also how to write a readable code.
Always keep in mind that someone might be reading it in 2 years. 
Also not sure in what case cell can be nullable, if it can it is worth mentioning it in the variable name or in the comment 
```suggestion
      row.forEach((cell) => {
      const isEnemyTeam =  cell.attributes.team !== this.attributes.team 
      const hasHp = cell.attributes.hp > 0
        if (isEnemyTeam && hasHp) {
          enemyUnits.push(cell);
        }
```